packages/pages/src/dev/client/entry-client.tsx

```diff
@@ -36,16 +36,23 @@ const hydrate = async () => {
     },
   };
 
-  const { default: Component } = await template.getComponent();
+  const { default: Component, render } = await template.getComponent();
```

Comment:
this whole function is called "hydrate" Why do we even need to check the render and optionally return?

I'd hope this code path wouldn't get hit at all when just render is defined.